server/routes/updateGoal/updateGoalController.ts

```diff
@@ -60,14 +61,29 @@ export default class UpdateGoalController {
       const nextStepNumber = currentHighestStepNumber + 1
       const newStep: UpdateStepForm = { stepNumber: nextStepNumber, status: 'NOT_STARTED' }
       updateGoalForm.steps.push(newStep)
-      req.session.updateGoalForm = updateGoalForm
       // Redirect back to the Update Goal page with named anchor taking the user straight to the new step
       return res.redirect(`/plan/${prisonNumber}/goals/${goalReference}/update#steps[${nextStepNumber - 1}][title]`)
     }
 
+    return res.redirect(`/plan/${prisonNumber}/goals/${goalReference}/update/review`)
+  }
+
+  getReviewUpdateGoalView: RequestHandler = async (req, res, next): Promise<void> => {
+    const { prisonerSummary } = req.session
+    const { updateGoalForm } = req.session
+
+    const view = new ReviewView(prisonerSummary, updateGoalForm)
+    res.render('pages/goal/update/review', { ...view.renderArgs })
+  }
+
+  submitReviewUpdateGoal: RequestHandler = async (req, res, next): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { updateGoalForm } = req.session
+
     const updateGoalDto = toUpdateGoalDto(updateGoalForm)
     await this.educationAndWorkPlanService.updateGoal(prisonNumber, updateGoalDto, req.user.token)
-    // TODO - RR-188 - handle API error response when updating goal
+
+    req.session.updateGoalForm = undefined
```

Comment:
💯 Yes, good work remembering / sussing out that we need to clear the form from the session once we are done 👍 

Not only is it a clean and tidy thing to do (we only want to hold things in the session for as long as we need them, otherwise the session could get massive); but also it could cause weird bugs. For example, if we've left an UpdateGoalForm in the session; then when someone next hits the Update Goal page (either by clicking Update from the Overview page, or perhaps using browser back navigation, or bookmarking URLs etc); then your new `checkUpdateGoalFormExistsInSession` request handler will kick in and will pass! Which might result in showing/editing the wrong Goal !!

Reply:
Can't take credit for this one! Think its something you did for Update a goal